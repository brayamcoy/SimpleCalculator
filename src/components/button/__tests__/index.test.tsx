import React from "react";
import { Text, Pressable } from "react-native";
import renderer, { act } from "react-test-renderer";
import ButtonCalculator, { ButtonProps } from "../index";

const defaultProps: ButtonProps = {
  children: "1",
  operator: false,
  onPress: jest.fn(),
};

describe("ButtonCalculator component", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<ButtonCalculator {...defaultProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with operator", () => {
    const props: ButtonProps = {
      ...defaultProps,
      operator: true,
    };
    const tree = renderer.create(<ButtonCalculator {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("calls onPress function on button press", () => {
    const tree = renderer.create(<ButtonCalculator {...defaultProps} />);
    act(() => {
      const pressable = tree.root.findByType(Pressable);
      pressable.props.onPress();
      // tree.root.findByType(Pressable).props.onPress();
    });

    expect(defaultProps.onPress).toHaveBeenCalledWith(defaultProps.children);
  });

  it("calls onPress function on button long press", () => {
    const tree = renderer.create(<ButtonCalculator {...defaultProps} />);
    act(() => {
      const pressable = tree.root.findByType(Pressable);
      pressable.props.onLongPress();
    });
    expect(defaultProps.onPress).toHaveBeenCalledWith(defaultProps.children);
  });

  it("Handles hover correctly", () => {
    const buttonCalculator = renderer.create(
      <ButtonCalculator onPress={() => {}}>1</ButtonCalculator>
    );
    act(() => {
      buttonCalculator.root.findByType(Pressable).props.onPressIn();
    });
    expect(buttonCalculator.toJSON()).toMatchSnapshot();
    act(() => {
      buttonCalculator.root.findByType(Pressable).props.onPressOut();
    });
    expect(buttonCalculator.toJSON()).toMatchSnapshot();
  });

  it("Should have correct style for operator button", () => {
    const component = renderer.create(
      <ButtonCalculator children="+" operator={true} onPress={jest.fn()} />
    );
    expect(component.root.findByType(Text).props.style).toHaveProperty(
      "color",
      "#FFF"
    );
  });
});
