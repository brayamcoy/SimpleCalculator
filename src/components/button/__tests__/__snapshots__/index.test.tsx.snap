// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonCalculator component Handles hover correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "borderRadius": 5,
        "padding": 35,
      },
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.1)",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#FFF",
        "fontSize": 16,
        "fontWeight": "500",
      }
    }
  >
    1
  </Text>
</View>
`;

exports[`ButtonCalculator component Handles hover correctly 2`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "borderRadius": 5,
        "padding": 35,
      },
      false,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#FFF",
        "fontSize": 16,
        "fontWeight": "500",
      }
    }
  >
    1
  </Text>
</View>
`;

exports[`ButtonCalculator component renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "borderRadius": 5,
        "padding": 35,
      },
      false,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#FFF",
        "fontSize": 16,
        "fontWeight": "500",
      }
    }
  >
    1
  </Text>
</View>
`;

exports[`ButtonCalculator component renders correctly with operator 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "borderRadius": 5,
        "padding": 35,
      },
      false,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#FFF",
        "fontSize": 22,
      }
    }
  >
    1
  </Text>
</View>
`;
